.github/workflows/release.yaml

```diff
@@ -0,0 +1,87 @@
+name: release
+on:
+  push:
+    tags:
+      - 'v*'
+jobs:
+  create-release:
+    name: create-release
+    runs-on: ubuntu-latest
+    outputs:
+      upload_url: ${{ steps.release.outputs.upload_url }}
+    steps:
+      - name: Create GitHub release
+        id: release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: ${{ github.ref }}
+          draft: true
```

Comment:
Can we have this be based on whether the tag is a pre-release? I think we can have a step that introspects the tag and then sets the variable depending on the format.

If I don't want to have to edit the job definition as well as create a tag, it should be one or the other I think, and I would prefer just to create a tag.

Reply:
Hmm... I think I explained this poorly. What I meant was that every release will be marked as draft (only those with write permissions will see the release). To make the release public you'll just have to go to _Releases_ and edit the newly created release and click on _publish_.

There's no job definition to edit here, it still works by just creating the tag. The only difference from how you currently do releases is that now your release will be already created with all the assets, so you can have a look at the built binaries before you publish the release.

Reply:
Oh that's really nice, thank you!